tests/Feature/EditPageTest.php

```diff
@@ -0,0 +1,191 @@
+{
+    use DatabaseMigrations;
+
+    private $centre;
+    private $user;
+    private $registration;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->centre = factory(App\Centre::class)->create();
+
+        // Create a User
+        $this->user =  factory(App\User::class)->create([
+            "name"  => "test user",
+            "email" => "testuser@example.com",
+            "password" => bcrypt('test_user_pass'),
+            "centre_id" => $this->centre->id,
+        ]);
+
+        // make centre some registrations
+        $this->registration = factory(App\Registration::class)->create([
+            "centre_id" => $this->centre->id,
+        ]);
+    }
+
+    /** @test */
+    public function itShowsAPrimaryCarerInput()
+    {
+        $pri_carer = $this->registration->family->carers->first();
+        $this->actingAs($this->user)
+            ->visit(URL::route('service.registration.edit', [ 'id' => $this->registration ]))
+            ->seeElement('input[name="carer"][value="'. $pri_carer->name .'"]')
+        ;
+    }
+
+    /** @test */
+    public function itShowsASecondaryCarerInput()
+    {
+        $this->actingAs($this->user)
+            ->visit(URL::route('service.registration.edit', [ 'id' => $this->registration ]))
+            ->seeElement('input[name="carer_adder_input"]')
+            ->seeElement('button[id="add-dob"]')
+        ;
+    }
+
+    /** @test */
+    public function itShowsAListOfSecondaryCarers()
+    {
+        // Clear the carers
+        $this->registration->family->carers()->delete();
+        // Make 4 more
+        $new_carers = factory(App\Carer::class, 4)->make();
+        // Add to Family
+        $this->registration->family->carers()->saveMany($new_carers);
+
+        // Get the carers again
+        $carers = $this->registration->family->carers;
+
+        // Knock the first one off
+        $carers->shift();
+
+        // There should be 3...
+        $this->assertTrue($carers->count() == 3);
+
+        // Find the edit page
+        $this->actingAs($this->user)
+            ->visit(URL::route('service.registration.edit', [ 'id' => $this->registration ]))
+        ;
+        // See the names in the page
+        foreach ($carers as $sec_carer) {
+             $this->see($sec_carer->name)
+                 ->seeElement('input[type="hidden"][value="'. $sec_carer->name .'"]')
+                 ;
+        }
+    }
+
+
+    /** @test */
+    public function itShowsAChildInputComplex()
+    {
+        $this->actingAs($this->user)
+            ->visit(URL::route('service.registration.edit', [ 'id' => $this->registration ]))
+            ->seeElement('input[name="dob-month"]')
+            ->seeElement('input[name="dob-year"]')
+            ->seeElement('button[id="add-dob"]')
+        ;
+    }
+
+    /** @test */
+    public function itShowsAListOfChildren()
+    {
+        // Clear the children
+        $this->registration->family->children()->delete();
```

Comment:
Why clear kids?

Reply:
Same reason. I want a known number of children for this test, not a random gaggle